src/main/java/de/retest/web/selenium/ByRetestId.java

```diff
@@ -0,0 +1,63 @@
+package de.retest.web.selenium;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.SearchContext;
+import org.openqa.selenium.WebElement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.retest.ui.descriptors.Element;
+import de.retest.ui.descriptors.RootElement;
+import de.retest.ui.diff.Alignment;
+
+public class ByRetestId extends By implements Serializable {
+
+	private static final long serialVersionUID = -3787115401615364934L;
+
+	private static final Logger logger = LoggerFactory.getLogger( ByRetestId.class );
+
+	private final String retestId;
+
+	public ByRetestId( final String retestId ) {
+		this.retestId = retestId;
+		if ( StringUtils.isBlank( retestId ) ) {
```

Comment:
Should use `RetestIdUtil#validate( String, IdentifyingAttributes)`, which already throws proper exceptions. (Or at least a `TODO` if not available as of now.)